src/core/include/openvino/core/node.hpp

```diff
@@ -207,6 +207,7 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
     virtual bool evaluate_upper(ov::TensorVector& output_values) const;
     virtual bool evaluate_symbol(TensorSymbolVector& output_symbols) const;
 
+    virtual bool can_constant_fold(const OutputVector& inputs_values);
```

Comment:
Do we need this additional member as public API?
It looks like this function can be internal in required operators as part of `constant_fold` and `constnat_fold` used instead.

Reply:
I had a talk with @jane-intel - she agreed with having a new public method.

Reply:
Sure. 

Optional suggestion: 
If we got additional member the change could look like:
```cpp
virtual bool can_constant_fold(const OutputVector& inputs_values);

protected:
   virtual void fold(OutputVector& output_values, const OutputVector& inputs_values);
 ```
 
 Then base class implementation of `constant_fold` could look like:
 ```cpp
 {
     if (can_constant_fold(input_values)){
         fold(output_values, input_values)
         return true;
     } else{
        return false;
     }
 }
 ```
 
I think the main algorithm will be same for most nodes but developer will have more flexibility in implementation as can override validation, folding implementation or the whole algorithm and still only two members are exposed as public API